### PR TITLE
[Graviton][build] Patch-Update PT1.12 TF2.9

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Small Patch update for PT1.12 and TF2.9 Graviton images

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -280,7 +280,9 @@ def get_dockerfile_path_for_image(image_uri):
 
 def get_expected_dockerfile_filename(device_type, image_uri):
     if is_covered_by_ec2_sm_split(image_uri):
-        if is_ec2_sm_in_same_dockerfile(image_uri):
+        if "graviton" in image_uri:
```
Comment (shantanutrip) [inline test/test_utils/__init__.py:283]: Thanks for adding this. Can we do the following:
1. Add logic in method `is_covered_by_ec2_sm_split` [link](https://github.com/aws/deep-learning-containers/blob/5ae80ebc2ad3f6723b53f0ab7bf8fe8172851681/test/test_utils/__init__.py#L455) to return `False` if the image is `Graviton`
2. Add a condition after [this](https://github.com/aws/deep-learning-containers/pull/2282/files#diff-5d5511865c124cd43c10b4a8f75f026de33c01632acdee7f0fbc849ae5498eb1R299) for Graviton images

Comment (xncqr) [reply test/test_utils/__init__.py:283]: I think we still want "is_covered_by_ec2_sm_split" logic to remain as an option for Graviton SageMaker images. if setting it to false when graviton is used, would pipeline not fine the Graviton Sagemaker images?

**Comment by arjkesh** [comment]
LGTM

File: pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu [hunk 2]
```diff
@@ -59,6 +60,7 @@ RUN apt-get update \
     wget \
     unzip \
     zlib1g-dev \
+ && apt-get autoremove -y \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu:63]: Do you know what packages this is removing? The issue with doing this on an existing version stream (as opposed to a new version) is that we may be removing dependencies that, while not used by other applications, may be used by customers. I suggest removing this line unless we can reliably predict the behavior of packages removed - esp for an image stream that has already been released.

Comment (xncqr) [reply pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu:63]: Much of this is for security hardening as it would remove unused packages with possible vulnerabilities. If customers are needing a packages, we would add it to the install list explicitly (can be seen in the current install) which would not be selected during the autoremove. Will do a build and see what the autoremove is removing in this case and will comment here. Would rather keep this.

Comment (xncqr) [reply pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu:63]: checked both images with the option disabled and manually executed autoremove. no packages were slated for this operation as this time. I feel this is safe to keep in the images.

**Review by arjkesh — APPROVED** [review]
approval to
